scripts/version_util.py

```diff
@@ -106,6 +106,10 @@ def main() -> None:
                 f"Version number in Cargo.toml ({cargo_version}) does not match tag version ({ref_version})"
             )
         print(f"Version numbers match: {cargo_version} == {ref_version}")
+    elif sys.argv[1] == "--expected_version":
+        # We always expect the version built to match what is in cargo.toml
+        # NOTE: what is in Cargo.toml may have been modified by an earlier call to `--patch-prerelease`
+        print(f"expected_version={cargo_version}")
```

Comment:
I find it a bit iffy that `expected_version` is set here, and then used in `python.yml`. A bit too magic, and breaks easily.

An alternative is something like this in `python.yml`:

```
printf "expected_version=$(scripts/version_util.py --cargo_version)" >> $GITHUB_ENV
```